// @turbopuffer/turbopuffer@1.21.0/resources/namespaces.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { path } from "../internal/utils/path.mjs";
import { NotFoundError } from "../error.mjs";
export class Namespace extends APIResource {
    /**
     * Delete namespace.
     */
    deleteAll(params = {}, options) {
        const { namespace = this._client.defaultNamespace } = params ?? {};
        return this._client.delete(path `/v2/namespaces/${namespace}`, options);
    }
    /**
     * Explain a query plan.
     */
    explainQuery(params = {}, options) {
        const { namespace = this._client.defaultNamespace, ...body } = params ?? {};
        return this._client.post(path `/v2/namespaces/${namespace}/explain_query`, { body, ...options });
    }
    /**
     * Signal turbopuffer to prepare for low-latency requests.
     */
    hintCacheWarm(params = {}, options) {
        const { namespace = this._client.defaultNamespace } = params ?? {};
        return this._client.get(path `/v1/namespaces/${namespace}/hint_cache_warm`, options);
    }
    /**
     * Get metadata about a namespace.
     */
    metadata(params = {}, options) {
        const { namespace = this._client.defaultNamespace } = params ?? {};
        return this._client.get(path `/v1/namespaces/${namespace}/metadata`, options);
    }
    /**
     * Issue multiple concurrent queries filter or search documents.
     */
    multiQuery(params, options) {
        const { namespace = this._client.defaultNamespace, ...body } = params;
        return this._client.post(path `/v2/namespaces/${namespace}/query?stainless_overload=multiQuery`, {
            body,
            ...options,
        });
    }
    /**
     * Query, filter, full-text search and vector search documents.
     */
    query(params = {}, options) {
        const { namespace = this._client.defaultNamespace, ...body } = params ?? {};
        return this._client.post(path `/v2/namespaces/${namespace}/query`, { body, ...options });
    }
    /**
     * Evaluate recall.
     */
    recall(params = {}, options) {
        const { namespace = this._client.defaultNamespace, ...body } = params ?? {};
        return this._client.post(path `/v1/namespaces/${namespace}/_debug/recall`, { body, ...options });
    }
    /**
     * Get namespace schema.
     */
    schema(params = {}, options) {
        const { namespace = this._client.defaultNamespace } = params ?? {};
        return this._client.get(path `/v1/namespaces/${namespace}/schema`, options);
    }
    /**
     * Check whether the namespace exists.
     */
    async exists() {
        try {
            await this.schema();
            return true;
        }
        catch (e) {
            if (e instanceof NotFoundError) {
                return false;
            }
            throw e;
        }
    }
    /**
     * Update namespace schema.
     */
    updateSchema(params = undefined, options) {
        const { namespace = this._client.defaultNamespace, schema } = params ?? {};
        return this._client.post(path `/v1/namespaces/${namespace}/schema`, { body: schema, ...options });
    }
    /**
     * Create, update, or delete documents.
     */
    write(params = {}, options) {
        const { namespace = this._client.defaultNamespace, ...body } = params ?? {};
        return this._client.post(path `/v2/namespaces/${namespace}`, { body, maxRetries: 6, ...options });
    }
}
